hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -944,13 +953,25 @@ public void startCommitWithTime(String instantTime) {
   }
 
   /**
-   * Completes a new commit time for a write operation (insert/update/delete/insert_overwrite/insert_overwrite_table) with specified action.
+   * Provides a new commit at the specified time with the provided action type.
+   * This is only for testing purposes to setup particular sequences of commits.
+   * @param instantTime the commit start time
+   * @param actionType the type of commit
    */
-  public void startCommitWithTime(String instantTime, String actionType) {
+  @VisibleForTesting
+  void startCommitWithTime(String instantTime, String actionType) {
```

Comment:
should we also deprecate `startCommitWithTime(String instantTime)` ?

Reply:
Yes, I have another branch that does this